TMDS_3/include/CGAL/IO/File_medit.h

```diff
@@ -865,53 +903,223 @@ output_to_medit(std::ostream& os,
+ *        Otherwise, only the cells `c` for which
+ *        `c->subdomain_index() != Subdomain_index()` are written}
+ *   \cgalParamType{Boolean}
+ *   \cgalParamDefault{`false`}
+ *   \cgalParamExtra{This parameter must be set to `true` for the file to be readable by `read_MEDIT()`.}
+ * \cgalParamNEnd
+ *
+ * \cgalParamNBegin{all_vertices}
+ *   \cgalParamDescription{If `true`, all the vertices in `t3` are written in `os`.
+ *                         Otherwise, only the vertices that belong to a cell `c` for which
+ *                         `c->subdomain_index() != Subdomain_index()` are written}
+ *   \cgalParamType{Boolean}
+ *   \cgalParamDefault{`true`}
+ *   \cgalParamExtra{If `all_cells` is `true, this parameter is ignored. It must be
+ *                   set to `false` for the file to be readable by `read_MEDIT()`.}
+ * \cgalParamNEnd
+ *
+ * \cgalParamNBegin{rebind_labels}
+ *  \cgalParamDescription{if `true`, labels of cells are rebinded into `[1..nb_of_labels]`}
+ *  \cgalParamType{Boolean}
+ *  \cgalParamDefault{`false`}
+ * \cgalParamNEnd
+ *
+ * \cgalParamNBegin{show_patches}
+ *  \cgalParamDescription{if `true`, patches are labeled with different labels than
+ *                        cells. If `false`, each surface facet is written twice,
+ *                        using the label of each adjacent cell.}
+ *  \cgalParamType{Boolean}
+ *  \cgalParamDefault{`true`}
+ * \cgalParamNEnd
+ * \cgalNamedParamsEnd
+ *
+ * \see \ref IOStreamMedit
+ */
+template<typename T3,
+         typename CornerIndex,
+         typename CurveIndex,
+         typename NamedParameters>
+void write_MEDIT(std::ostream& os,
+  const CGAL::Mesh_complex_3_in_triangulation_3<T3, CornerIndex, CurveIndex>& c3t3,
+  const NamedParameters& np)
+{
+  return write_MEDIT(os, c3t3.triangulation(), np);
+}
+
+template<typename T3,
+         typename CornerIndex,
+         typename CurveIndex,
+         typename NamedParameters>
+void write_MEDIT(std::ostream& os,
+    const CGAL::Mesh_complex_3_in_triangulation_3<T3, CornerIndex, CurveIndex>& c3t3)
+{
+  return write_MEDIT(os, c3t3, parameters::all_default());
+}
+
+/**
+ * @ingroup PkgTMDS3IOFunctions
+ * @brief reads a mesh complex written in the medit (`.mesh`) file format.
+ *   See \cgalCite{frey:inria-00069921} for a comprehensive description of this file format.
+ * @tparam T3 can be instantiated with any 3D triangulation of \cgal provided that its
+ *  vertex and cell base class are models of the concepts `MeshVertexBase_3` and `MeshCellBase_3`,
+ *  respectively.
+ *
+ * @param in the input stream
+ * @param t3 the triangulation
+ *
+ * This function reads the data about vertices, surface facets and
+ * triangulation cells from `in`, and builds a valid `T3` from it.
+ *
+ * Note that a valid 3D triangulation of \cgal must have a valid
+ * data structure (see `TriangulationDataStructure_3 `),
+ * positively oriented cells,
+ * and cover the geometric convex hull of all points in `t3`.
+ */
```

Comment:
The returned value (`true` of `false`) should be documented. In particular, it should be documented what happens if the input file is not a "a valid 3D triangulation of \cgal": does it crash, or does it return `false` without changing `t3`?